### PR TITLE
Plot numeric and discrete sensors on the same plot view using two y axis feature

### PR DESCRIPTION
**Pull request by bngcebetsha**
## This PR

There is a benefit for combined discrete and value sensor graphs.  For example, with pedestal tilt work it would be beneficial to see the correlation between tiltmeter read errors and `struct_tilt_x` and `struct_tilt_y` values. See [MKT-148](https://skaafrica.atlassian.net/browse/MKT-148) for more details.

In this PR, we conclude the delivering of this feature, we had broken it down to small steps namely:
1. Plot two y-axes on sensor-graph, one on the left and one on the right
2. Plot numeric sensors on left and discrete sensors on the right y-axis
3. Retain current plot when searching for a different sensor type than the one currently plotted
4. Plot both numeric and discrete sensors on the same graph

This PR does part 4 of the list above and combines all the work from the other steps.

## Work still in progress
1. Retain sensor names stored and displayed as "chips" when changing sensor types, they are erased when we make a switch although the graphs is as desired
2. Rework `clearData` JS method to maintain it's current functionality of "clearing the entire plot"

sensor-graph

#### NB: Reviewer
No need to review this list of `Files changed`
- `Jenkinsfile`
- `Makefile`
- `README.md`
- `dist/app.full.min.1653367884635.css`
- `dist/app.full.min.1653367884635.js`
- `dist/app.full.min.1658324831323.css`
- `dist/app.full.min.1658324831323.js`
- `dist/index.html`
- `dist/kat-version.txt`
- `dist/version.txt`
- `gulpfile.js`
- `misc/check_pr_state.py`
- `package.json`
- `yarn.lock`

JIRA: [MT-3056](https://skaafrica.atlassian.net/browse/MT-3056)

### CODE REVIEW

File: app/d3/multi-line-chart.js [hunk 1]
```diff
@@ -224,8 +241,9 @@ angular.module('katGui.d3')
             };
 
             scope.clearFunction = function() {
-                scope.options.yAxisValues = null;
-                scope.nestedData = [];
+                // scope.options.yAxisValues = null;
+                // scope.options.yAxisRightValues = null;
+                //scope.nestedData = [];
```
Comment (mmphego) [inline app/d3/multi-line-chart.js:246]: Perhaps, remove this code if it doesn't add value.

File: app/d3/multi-line-chart.js [hunk 2]
```diff
@@ -333,12 +373,8 @@ angular.module('katGui.d3')
 
                 // set the ranges
                 x = d3.time.scale.utc().range([0, width]);
-                y = null;
-                if (scope.options.discreteSensors) {
-                    y = d3.scale.ordinal().rangePoints([height, 0]);
-                } else {
-                    y = d3.scale.linear().range([height, 0]);
-                }
+                yRight = d3.scale.ordinal().rangePoints([height, 0]);
```
Comment (mmphego) [inline app/d3/multi-line-chart.js:376]: 👍🏽

**Review by mmphego — COMMENTED** [review]
Logic looks good to me, please check CI

File: app/d3/multi-line-chart.js (hunk 1 above)
Comment (bngcebetsha) [reply app/d3/multi-line-chart.js:246]: WIP

**Comment by rohanschwartz** [comment]
@bngcebetsha has this been deployed to devkaroo?

**Comment by bngcebetsha** [comment]
> @bngcebetsha has this been deployed to devkaroo?

for some strange reason I do not get email notifications from guthub anymore....No @rohanschwartz we have just one last bug fix that is on this PR https://github.com/ska-sa/katgui/pull/636 that @tockards is working on.

**Pull request merged**